clang/lib/Driver/Driver.cpp

```diff
@@ -4794,6 +4800,10 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
     // archive unbundling for Windows.
     if (!isStaticArchiveFile(LA))
       continue;
+    // FPGA AOCX files are archives, but we do not want to unbundle them here
+    // as they have already been unbundled and processed for linking.
+    if (hasFPGABinary(C, LA.str(), types::TY_FPGA_AOCX))
+      continue;
```

Comment:
Should the check insertion also be covered by a `-###` test? (I could've overlooked one)

Reply:
Added a `-NOT` check to be sure an additional unbundle step is not done.

Reply:
Yet, are separate bundler invocations to facilitate the AOCX check immune to the negative line?..

Reply:
Looks to be.  The `NOT` checks are beyond the initial AOCX checks and the unbundle we want to capture.  Testing doesn't trigger so it looks OK.

Reply:
Right, thanks. I believe the negative check is sufficient then - the fact that bundler runs are used to facilitate it doesn't have much testing value if we know the check took effect.